x-pack/plugins/uptime/public/pages/monitor.tsx

```diff
@@ -15,6 +15,20 @@ import { useMonitorId, useUptimeTelemetry, UptimePage } from '../hooks';
 import { MonitorCharts } from '../components/monitor';
 import { MonitorStatusDetails, PingList } from '../components/monitor';
 import { getDynamicSettings } from '../state/actions/dynamic_settings';
+import { Ping } from '../../common/runtime_types/ping';
+
+const isAutogeneratedId = (id: string) => {
+  return /^auto-(icmp|http|tcp)-OX[A-F0-9]{16}-[a-f0-9]{16}/.test(id);
+};
+
+// For auto Gen Id we will display url since ID looks too ugly in the UI
+// We will plan to deprecate it after 8.0.0 once is required in heartbeat config
+const getIdToDisplay = (monId: string, selectedMonitor: Ping | null) => {
```

Comment:
I think `getPageTitle` might be a more accurate description of what this function does, since the URL isn't really an ID value.